### PR TITLE
dev/run: print log dir on startup

### PR DESCRIPTION
**Pull request by alxndrsn**
## Overview

When running couch from source, logs do not appear on stdout.  The actual location of the logs is not documented.  It would be helpful to make log location clear.

### Before

```
$ ./dev/run
[ * ] Setup environment ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Ensure Erlang boot script exists ... ok
[ * ] Prepare configuration files ... ok
[ * ] Start node node1 ... ok
[ * ] Start node node2 ... ok
[ * ] Start node node3 ... ok
[ * ] Check node at http://127.0.0.1:15984/ ... ok
[ * ] Check node at http://127.0.0.1:25984/ ... ok
[ * ] Check node at http://127.0.0.1:35984/ ... ok
[ * ] Running cluster setup ... ok
[ * ] Developers cluster is set up at http://127.0.0.1:15984.
Admin username: root
Password: rfb7saVr
Time to hack! ...
```

### After

```
$ ./dev/run
[ * ] Setup environment ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Ensure Erlang boot script exists ... ok
[ * ] Prepare configuration files ... ok
[ * ] Start node node1 ... ok
[ * ] Start node node2 ... ok
[ * ] Start node node3 ... ok
[ * ] Check node at http://127.0.0.1:15984/ ... ok
[ * ] Check node at http://127.0.0.1:25984/ ... ok
[ * ] Check node at http://127.0.0.1:35984/ ... ok
[ * ] Running cluster setup ... ok
[ * ] Developers cluster is set up at http://127.0.0.1:15984.
Admin username: root
Password: asdf1234
Logs: /couchdb/dev/logs
Time to hack! ...
```

## Testing recommendations

Run `./dev/run`.

## Related Issues or Pull Requests

None known.

## Checklist

- [x] Code is written and works correctly
- ~Changes are covered by tests~
- [x] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- ~Documentation changes were made in the `src/docs` folder~
- ~Documentation changes were backported (separated PR) to affected branches~

Removed checklist items as I think this change is self-documenting, and does not require tests.

### CODE REVIEW

File: dev/run [hunk 1]
```diff
@@ -93,7 +93,12 @@ def main():
         if ctx["cmd"]:
             run_command(ctx, ctx["cmd"])
         else:
-            join(ctx, cluster_port(ctx, ctx["node_number_seed"]), *ctx["admin"])
+            join(
+                ctx,
+                cluster_port(ctx, ctx["node_number_seed"]),
+                *ctx["admin"],
+                os.path.join(ctx["devdir"], "logs"),
```
[the remark this answers is not in the document]
Comment (big-r81) [reply dev/run:100]: Yeah, I think a relative path would make more sense here!

**Comment by alxndrsn** [comment]
I've updated to output a relative path instead of absolute.

**Comment by rnewson** [comment]
I think it's a useful thing to print, but I don't like adding yet another construction of "devdir" + "logs". I suggest we tidy this up first, and have a function that calculates the log dir from `ctx`, use that everywhere, and then ensure it is logged somewhere (once).

Something like https://github.com/apache/couchdb/compare/run-logs-dir-2?expand=1 for example. 

I quite like the absolute path but relative is also fine, just as long as it works on all platforms (i.e, `./` as in the PR currently is not going to fly on Windows).

**Comment by big-r81** [comment]
Short summary:

How do we want to bring these two solutions together?

1. This PR: 
Print relative directory and manual joining of the path parts.

2. Bob's idea: 
Generic function for getting the log dir and print it as absolute path.

I tried Bob's idea on Windows an macOS:
```
Windows:
[ * ] Setup log dir: C:\Users\couchdb\Documents\couchdb-otp25\dev\logs ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Ensure Erlang boot script exists ... ok

macOS:
[ * ] Setup log dir: /Users/big-r/Documents/Developer/CouchDB/couchdb/dev/logs ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Ensure Erlang boot script exists ... ok
```

After a few days have passed, I like Bob's (@rnewson) generic dev dir function a little bit more. It's cleaner, but with abs paths. What do you think Nick (@nickva)?

**Comment by nickva** [comment]
@big-r81 Let's go with Bob's idea then, the preference for relative paths was a minor preference.  Can always redact it away when pasting anywhere public.

**Comment by rnewson** [comment]
https://github.com/apache/couchdb/pull/5136 then?

**Comment by big-r81** [comment]
> #5136 then?

Yes 👍.

**Comment by big-r81** [comment]
@alxndrsn 
Thanks for your initial idea and contribution to make CouchDB better. 🙏

I’m closing this in favor of #5136.

**Pull request closed without merging**